ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoice.java

```diff
@@ -927,4 +931,25 @@ private Optional<Boolean> getOptimisticSyncing() {
   public interface OptimisticHeadSubscriber {
     void onOptimisticHeadChanged(boolean isHeadOptimistic);
   }
+
+  // Implements `validate_inclusion_lists` added in EIP-7805 - consensus/fork-choice
+  public void validateInclusionLists(
+      final UpdatableStore store,
+      final List<Transaction> inclusionListTransactions,
+      final ExecutionPayload executionPayload) {
+    final SpecConfigEip7805 eip7805 = spec.getGenesisSpec().getConfig().toVersionEip7805().get();
+    final int maxTransactionPerInclusionList = eip7805.getMaxTransactionsPerInclusionList();
+    final int inclusionListCommitteeSize = eip7805.getInclusionListCommitteeSize();
+
+    if (inclusionListTransactions.size()
+        > maxTransactionPerInclusionList * inclusionListCommitteeSize) {
+      throw new IllegalStateException("Inclusion list has too many transactions");
```

Comment:
I'm adding some `InclusionListImportResult` classes in my next PR and I'm think it would better to return those instead of throwing